### PR TITLE
move dns api key to a variable for govcloud

### PR DESCRIPTION
**Pull request by enxebre**
move dns api key to a variable for govcloud https://github.com/coreos/tectonic-installer/pull/2836

### CODE REVIEW

**Comment by cpanato** [comment]
@enxebre  some errors:

```
00:13:16.727 2018/01/26 10:15:53 [ERROR] 5 problems:
00:13:16.727 
00:13:16.727 - module 'ingress_certs': unknown variable referenced: 'tectonic_govcloud_private_endpoints'; define it with a 'variable' block
00:13:16.727 - module 'bootkube': unknown variable referenced: 'tectonic_govcloud_private_endpoints'; define it with a 'variable' block
00:13:16.727 - module 'kube_certs': unknown variable referenced: 'tectonic_govcloud_private_endpoints'; define it with a 'variable' block
00:13:16.727 - module 'masters': unknown variable referenced: 'tectonic_govcloud_public_endpoints'; define it with a 'variable' block
00:13:16.727 - module 'tectonic': unknown variable referenced: 'tectonic_govcloud_private_endpoints'; define it with a 'variable' block
```

**Comment by cpanato** [comment]
retest this please

File: tests/rspec/lib/govcloud_vpc.rb [hunk 1]
```diff
@@ -35,7 +35,9 @@ def export_tfvars
       'TF_VAR_tectonic_govcloud_external_vpc_id' => @vpc_id,
       'TF_VAR_tectonic_govcloud_external_master_subnet_ids' => @master_subnet_ids,
       'TF_VAR_tectonic_govcloud_external_worker_subnet_ids' => @worker_subnet_ids,
-      'TF_VAR_tectonic_govcloud_dns_server_ip' => @vpc_dns
+      'TF_VAR_tectonic_govcloud_dns_server_ip' => @vpc_dns,
+      'TF_VAR_tectonic_govcloud_dns_server_api_url' => @dns_api_url,
+      'TF_VAT_tectonic_govcloud_dns_server_api_key' => 'tectonicgov'
```
Comment (cpanato) [inline tests/rspec/lib/govcloud_vpc.rb]: typo here. will fix

File: platforms/govcloud/main.tf [hunk 2]
```diff
@@ -6,6 +6,11 @@ provider "aws" {
 
 data "aws_availability_zones" "azs" {}
 
+locals {
```
Comment (squat) [inline platforms/govcloud/main.tf]: Same as #2836: @enxebre here you are hardcoding `private_endpoints` to `true` and `public_endpoints` to `false` but you are keeping all of the ternary expressions. Why not just simplify this and eliminate all of the ternarys? It makes it much easier to read. These changes are in VCS so we don't need to keep this around. If you really need to keep it around even if the ternarys do nothing, then please add a comment

**Comment by cpanato** [comment]
@enxebre lets follow up the @squat  comments in a separate PR and merge this one now for track-1.

cc @sym3tri

**Pull request merged**